src/cmd-buildextend-installer

```diff
@@ -126,24 +126,38 @@ initrd_ignition_padding = 256 * 1024
 
 
 # https://www.kernel.org/doc/html/latest/admin-guide/initrd.html#compressed-cpio-images
-def mkinitrd_pipe(tmproot, destf):
+def mkinitrd_pipe(tmproot, destf, compress=True):
+    # The kernel requires that uncompressed appended archives start on a
+    # 4-byte boundary.  If there's misalignment, it stops unpacking and
+    # says:
+    #   Initramfs unpacking failed: invalid magic at start of compressed archive
+    # https://github.com/torvalds/linux/blob/47ec5303/init/initramfs.c#L463
```

Comment:
A more human-friendly source: https://www.kernel.org/doc/Documentation/early-userspace/buffer-format.txt

Reply:
I wouldn't mind a link to both.